clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3569,6 +3557,69 @@ protected enum DelegateConsumerRecordResult {
+      int versionNumber,
+      PartitionConsumptionState partitionConsumptionState,
+      KafkaMessageEnvelope kafkaMessageEnvelope,
+      LeaderProducedRecordContext leaderProducedRecordContext) {
+    long afterProcessingRecordTimestampMs = System.currentTimeMillis();
+    if (!isUserSystemStore() && isHybridMode() && partitionConsumptionState.hasLagCaughtUp()
+        && (partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.LEADER
+            || partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.STANDBY)) {
+      if (partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.LEADER
+          && leaderProducedRecordContext == null) {
+        LOGGER.error(
+            "LeaderProducedRecordContext is null when PCS state is LEADER for storeName: {},"
+                + " versionNumber: {}, partition: {}",
+            storeName,
+            versionNumber,
+            partitionConsumptionState.getPartition());
+        return;
+      }
+      if (partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.STANDBY
+          && kafkaMessageEnvelope.getLeaderMetadataFooter() == null) {
+        LOGGER.error(
+            "LeaderMetadataFooter is null when PCS state is STANDBY for storeName: {},"
+                + " versionNumber: {}, partition: {}",
+            storeName,
+            versionNumber,
+            partitionConsumptionState.getPartition());
+        return;
+      }
+      boolean hasCorrespondingUpstreamMessage = false; // No point in recording E2E latency if there is no upstream
+      if (partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.LEADER) {
+        hasCorrespondingUpstreamMessage = leaderProducedRecordContext.hasCorrespondingUpstreamMessage();
+      } else if (partitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.STANDBY
+          && kafkaMessageEnvelope.getLeaderMetadataFooter() != null) {
+        hasCorrespondingUpstreamMessage = kafkaMessageEnvelope.getLeaderMetadataFooter().getUpstreamOffset() > 0;
+      }
+      if (hasCorrespondingUpstreamMessage) {
+        long producerTimestamp = (leaderProducedRecordContext == null)
+            ? kafkaMessageEnvelope.producerMetadata.messageTimestamp
+            : leaderProducedRecordContext.getProducedTimestampMs();
+        if (producerTimestamp <= 0) {
+          LOGGER.warn(
+              "Illegal timestamp for storeName: {}, versionNumber: {}, partition: {}, "
+                  + "leaderProducedRecordContext: {}, leaderMetadataFooter: {}",
+              storeName,
+              versionNumber,
+              partitionConsumptionState.getPartition(),
+              leaderProducedRecordContext == null ? "NA" : leaderProducedRecordContext,
+              kafkaMessageEnvelope.getLeaderMetadataFooter() == null
+                  ? "NA"
+                  : kafkaMessageEnvelope.getLeaderMetadataFooter());
```

Comment:
Same here please - please use redundant log filtering; and this time we cannot simply use the message since offset is in the message; you can break the message into two parts, where the first part only has store name, version and partition, then use the first part as filtering input.

I am worried that the new logs would end up flooding the log file.

Reply:
ok. will use the filtering